client/tests/tests/events.rs

```diff
@@ -6,49 +6,100 @@ use std::{
 };
 
 use eyre::Result;
+use iroha_client::client::Client;
 use iroha_core::config::Configuration;
 use iroha_data_model::prelude::*;
-use test_network::{Peer as TestPeer, *};
+use test_network::*;
 
-#[test]
-fn transaction_event_should_be_sent_after_it_is_committed() -> Result<()> {
-    let (_rt, _peer, mut iroha_client) = <TestPeer>::start_test_with_runtime();
-    let pipeline_time = Configuration::pipeline_time();
-
-    // Given
-    thread::sleep(pipeline_time);
+const PEER_COUNT: usize = 7;
 
+#[test]
+fn transaction_event_should_be_sent_to_all_peers_from_all_peers() -> Result<()> {
     let asset_definition_id = AssetDefinitionId::new("xor", "wonderland");
     let create_asset = RegisterBox::new(IdentifiableBox::AssetDefinition(
         AssetDefinition::new_quantity(asset_definition_id).into(),
     ));
-    let committed_event_received = Arc::new(RwLock::new(false));
-    let committed_event_received_clone = committed_event_received.clone();
-    let client_clone = iroha_client.clone();
-    let _handle = thread::spawn(move || {
-        client_clone.loop_on_events(
-            EventFilter::Pipeline(PipelineEventFilter::by_entity(
-                PipelineEntityType::Transaction,
-            )),
-            |event| {
-                if let Ok(Event::Pipeline(event)) = event {
-                    //TODO: check transaction hash
-                    if event.entity_type == PipelineEntityType::Transaction
-                        && event.status == PipelineStatus::Committed
-                    {
-                        *committed_event_received_clone
-                            .write()
-                            .expect("Failed to acquire lock.") = true;
-                    }
-                }
-            },
-        );
-    });
-    thread::sleep(pipeline_time * 2);
-    //When
-    iroha_client.submit(create_asset)?;
-    thread::sleep(pipeline_time * 2);
-    //Then
-    assert!(*committed_event_received.read().unwrap());
+    test_with_instruction_and_status(create_asset, true)?;
+    let fail = FailBox::new("Failing transaction to test Rejected event.");
+    test_with_instruction_and_status(fail, false)
+}
+
+#[allow(clippy::needless_range_loop)]
+fn test_with_instruction_and_status(
+    instruction: impl Into<Instruction>,
+    should_be_committed: bool,
+) -> Result<()> {
+    let instruction: Instruction = instruction.into();
+    for submitting_peer in 0..PEER_COUNT {
+        let (_rt, network, _) =
```

Comment:
why aren't you using client returned here and create new one instead?

Reply:
That client is preconfigured to connect to specific peer. The test needs connection to different peers.